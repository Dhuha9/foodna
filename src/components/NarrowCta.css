@import url('https://fonts.googleapis.com/css2?family=Roboto&display=swap');

* {
    box-sizing: border-box;
}

.container{
  background-color: #F2F2F2;
  color: 000;
  display: flex;
  align-items: center;
  justify-content: space-around;
  margin: 0;
  font-family: 'Roboto', sans-serif;
  height: 240px;
  flex-wrap: wrap;
}

.content{
  display: flex;
  align-items: flex-start;
  justify-content: flex-start;
  flex-direction: column;
  color:#03B664;
  font-family: 'Roboto';
  font-size: 32px;
}

.content h4{
    font-family: 'Roboto';
    font-weight: bold;
    font-size: 32px;
    color: #000000;
    margin-bottom: 0px;
}

.buttons{
  display: flex;
  align-items: center;
  justify-content: flex-start;
  flex-direction: row;
}

