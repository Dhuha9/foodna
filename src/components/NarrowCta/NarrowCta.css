@import url('https://fonts.googleapis.com/css2?family=Roboto&display=swap');

* {
  box-sizing: border-box;
}

.NarrowCtaStyle .ctaContainer {
  background-color: #f2f2f2;
  color: 000;
  display: flex;
  align-items: center;
  justify-content: space-around;
  margin: 0;
  min-height: 240px;
  font-family: 'Roboto', sans-serif;
  flex-wrap: wrap;
}

.NarrowCtaStyle .content {
  display: flex;
  align-items: flex-start;
  justify-content: flex-start;
  flex-direction: column;
  color: #03b664;
  font-family: 'Roboto';
  font-size: 32px;
}

.NarrowCtaStyle .content h4 {
  font-family: 'Roboto';
  font-weight: bold;
  font-size: 32px;
  color: #000000;
  margin-bottom: 0px;
}

.NarrowCtaStyle .buttons {
  display: flex;
  align-items: center;
  justify-content: flex-start;
  flex-direction: row;
}

.NarrowCtaStyle .buttons button {
  margin-right: 10px;
}

@media only screen and (max-width: 700px) {
  /* For mobile screens */
  .NarrowCtaStyle .content {
    margin-bottom: 40px;
  }
  .NarrowCtaStyle .buttons {
    display: block;
    padding-top: 10px;
  }
  .NarrowCtaStyle .buttons button {
    margin-bottom: 20px;
  }
}
