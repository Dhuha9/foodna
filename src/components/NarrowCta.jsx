import React from 'react'
import './NarrowCta.css';
import CtaButton from './CtaButton';

var buttonStyle = {
    width: '240px',
    height: '60px',
    left: 'calc(50% - 240px/2 - 1px)',
    top: '256px',
    borderRadius:'5px',
    borderColor: '#000',
    fontSize: '18px',
    fontSamily: 'roboto',
    marginLeft: '20px',
    backgroundColor: '#F2F2F2'
}

export default function NarrowCta() {
    return (
        <div className="container">
            <div className="content">
                <h4>Want to help?</h4>
                <div>Donate now or contact us</div>
            </div>
            <div className="buttons">
                <CtaButton />
                <button
                style={buttonStyle}>
                    Contact Us
            </button>
            </div>
        </div>
    )
}
