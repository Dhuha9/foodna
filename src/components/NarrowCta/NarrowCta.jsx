import React from 'react';
import './NarrowCta.css';
import CtaButton from '../CtaButton/CtaButton';
import { Link } from 'react-router-dom';
import { useTranslation } from 'react-i18next';

var buttonStyle = {
  width: '240px',
  height: '60px',
  left: 'calc(50% - 240px/2 - 1px)',
  top: '256px',
  borderRadius: '5px',
  borderColor: '#000',
  fontSize: '18px',
  fontSamily: 'roboto',
  backgroundColor: '#F2F2F2',
};

export default function NarrowCta() {
  const { t } = useTranslation();

  return (
    <div className="NarrowCtaStyle">
      <div className="ctaContainer">
        <div className="content">
          <h4>{t('NarrowCtaComponent.title')}</h4>
          <p>{t('NarrowCtaComponent.title2')}</p>
        </div>

        <div className="buttons">
          <CtaButton />
          <Link to="/contact" style={buttonStyle}>
            <button style={buttonStyle}>
              {t('NarrowCtaComponent.contactUs')}
            </button>
          </Link>
        </div>
      </div>
    </div>
  );
}
