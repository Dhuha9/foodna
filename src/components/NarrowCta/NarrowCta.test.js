import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import NarrowCta from './NarrowCta';

it('renders when there is no NarrowCta', () => {
  const tree = renderer
    .create(
      <Router>
        <NarrowCta />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
