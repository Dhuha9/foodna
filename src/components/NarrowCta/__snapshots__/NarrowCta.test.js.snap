// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when there is no NarrowCta 1`] = `
<div
  className="NarrowCtaStyle"
>
  <div
    className="ctaContainer"
  >
    <div
      className="content"
    >
      <h4>
        NarrowCtaComponent.title
      </h4>
      <p>
        NarrowCtaComponent.title2
      </p>
    </div>
    <div
      className="buttons"
    >
      <a
        href="/donate"
        onClick={[Function]}
      >
        <button
          className="btn btn-default"
          style={
            Object {
              "backgroundColor": "#FFB701",
              "border": "none",
              "borderRadius": "5px",
              "color": "#000",
              "fontSamily": "roboto",
              "fontSize": "18px",
              "height": "60px",
              "left": "calc(50% - 240px/2 - 1px)",
              "top": "256px",
              "width": "240px",
            }
          }
        >
          CtaButton.title
        </button>
      </a>
      <a
        href="/contact"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "#F2F2F2",
            "borderColor": "#000",
            "borderRadius": "5px",
            "fontSamily": "roboto",
            "fontSize": "18px",
            "height": "60px",
            "left": "calc(50% - 240px/2 - 1px)",
            "top": "256px",
            "width": "240px",
          }
        }
      >
        <button
          style={
            Object {
              "backgroundColor": "#F2F2F2",
              "borderColor": "#000",
              "borderRadius": "5px",
              "fontSamily": "roboto",
              "fontSize": "18px",
              "height": "60px",
              "left": "calc(50% - 240px/2 - 1px)",
              "top": "256px",
              "width": "240px",
            }
          }
        >
          NarrowCtaComponent.contactUs
        </button>
      </a>
    </div>
  </div>
</div>
`;
